packages/status-js/src/request-client/request-client.ts

```diff
@@ -197,41 +245,48 @@ class RequestClient {
             return
           }
 
-          const signerPublicKey = `0x${compressPublicKey(
-            message.signerPublicKey
-          )}`
-
           // isSignatureValid
           if (isEncrypted(decodedCommunityDescription.tokenPermissions)) {
-            // const permission = Object.values(
-            //   decodedCommunityDescription.tokenPermissions
-            // ).find(
-            //   permission =>
-            //     permission.type ===
-            //     CommunityTokenPermission_Type.BECOME_TOKEN_OWNER
-            // )
-            // if (!permission) {
-            //   return
-            // }
-            // const criteria = permission.tokenCriteria[0]
-            // const contracts = criteria?.contractAddresses
-            // const chainId = Object.keys(contracts)[0]
-            // if (!chainId) {
-            //   return
-            // }
-            // // get client config based on chainId
-            // // get client
-            // const client = new EthereumClient(
-            //   `https://mainnet.infura.io/v3/${process.env.KEY}`
-            // )
-            // // call status contract for chainId
-            // const address = publicKeyToETHAddress(publicKey)
-            // // call contracts from previous call with address
-            // const ownerPublicKey = '0x0'
-            // if (ownerPublicKey !== signerPublicKey) {
-            //   return
-            // }
-          } else if (publicKey !== signerPublicKey) {
+            // todo?: zod
```

Comment:
this is not coming from protobufs?

Reply:
What isn't?

The protobufs fields are optional by design.